jsapp/js/account/organization/OrganizationSettingsRoute.tsx

```diff
@@ -1,7 +1,173 @@
+import KoboSelect from 'jsapp/js/components/common/koboSelect';
+
+// Stores, hooks and utilities
+import useWhenStripeIsEnabled from 'js/hooks/useWhenStripeIsEnabled.hook';
+import {OrganizationUserRole, useOrganizationQuery, usePatchOrganization} from 'js/account/organization/organizationQuery';
+import subscriptionStore from 'js/account/subscriptionStore';
+import envStore from 'js/envStore';
+import {getSimpleMMOLabel} from './organization.utils';
+
+// Constants and types
+import {ORGANIZATION_TYPES, type OrganizationTypeName} from 'jsapp/js/account/organization/organizationQuery';
+
+// Styles
+import styles from 'js/account/organization/organizationSettingsRoute.module.scss';
+
+/**
+ * Renders few fields with organization related settings, like name or website
+ * (with some logic in regards to their visibility). If user has necessary role,
+ * they can edit available fields.
+ */
 export default function OrganizationSettingsRoute() {
+  const orgQuery = useOrganizationQuery();
+  const [subscriptions] = useState(() => subscriptionStore);
+  const [isStripeEnabled, setIsStripeEnabled] = useState(false);
+  const patchOrganization = usePatchOrganization();
+
+  // All displayed fields
+  const [name, setName] = useState<string>('');
+  const [website, setWebsite] = useState<string>('');
+  const [orgType, setOrgType] = useState<OrganizationTypeName | null>(null);
+
+  useEffect(() => {
+    if (orgQuery.data) {
+      setName(orgQuery.data.name);
+      setWebsite(orgQuery.data.website);
+      setOrgType(orgQuery.data.organization_type);
+    }
+  }, [orgQuery.data]);
+
+  useWhenStripeIsEnabled(() => {
+    setIsStripeEnabled(true);
+  }, []);
+
+  const isUserAdminOrOwner = (
+    orgQuery.data?.request_user_role &&
+    [OrganizationUserRole.admin, OrganizationUserRole.owner]
+      .includes(orgQuery.data?.request_user_role)
+  );
+
+  function handleSave(e: React.FormEvent) {
+    e.preventDefault();
+    patchOrganization.mutateAsync({name, website});
+  }
+
+  function handleChangeName(newName: string) {
+    setName(newName);
+  }
+
+  function handleChangeWebsite(newWebsite: string) {
+    setWebsite(newWebsite);
+  }
+
+  const mmoLabel = getSimpleMMOLabel(
+    envStore.data,
+    subscriptionStore.activeSubscriptions[0],
+    false,
+    true
+  );
+  const mmoLabelLowercase = mmoLabel.toLowerCase();
+
+  if (orgQuery.isLoading) {
+    return <LoadingSpinner />;
+  }
+
+  let deletionMessage = t('To delete this ##team/org##, please contact the server administrator.')
+    .replaceAll('##team/org##', mmoLabelLowercase);
+  if (isStripeEnabled) {
+    deletionMessage = t("To delete this ##team/org##, you need to cancel your current ##plan name## plan. At the end of the plan period your ##team/org##'s projects will be converted to projects owned by your personal account.")
+    .replaceAll('##team/org##', mmoLabelLowercase)
+    .replace('##plan name##', subscriptions.planName);
+  }
```

Comment:
more of a linter thing, but don't use `let` and mutation if possible. Here one way to rewrite it
```suggestion
  let deletionMessage = isStripeEnabled
    ? t("To delete this ##team/org##, you need to cancel your current ##plan name## plan. At the end of the plan period your ##team/org##'s projects will be converted to projects owned by your personal account.")
      .replaceAll('##team/org##', mmoLabelLowercase)
      .replace('##plan name##', subscriptions.planName)
    : t('To delete this ##team/org##, please contact the server administrator.')
      .replaceAll('##team/org##', mmoLabelLowercase);
```
or 
```ts
  let deletionMessage = t(isStripeEnabled
    ? "To delete this ##team/org##, you need to cancel your current ##plan name## plan. At the end of the plan period your ##team/org##'s projects will be converted to projects owned by your personal account."
    : 'To delete this ##team/org##, please contact the server administrator.'
    )
      .replaceAll('##team/org##', mmoLabelLowercase)
      .replace('##plan name##', subscriptions.planName)
```